### PR TITLE
Reference id to data conversion

### PR DESCRIPTION
**Pull request by cscharf**
## Objective
Companion to the server change for reference id to be a data field (json blob) with a structured model for API submission through jslib. Need to capture 3 data points to send back through for flow and layout in addition to id.

### CODE REVIEW

File: src/models/domain/referenceEventData.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export class ReferenceEventData {
```
Comment (kspearrin) [inline src/models/domain/referenceEventData.ts:1]: Per server PR comments, this should probably be moved and renamed as a API request model.

File: src/models/request/registerRequest.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { KeysRequest } from './keysRequest';
 
 import { KdfType } from '../../enums/kdfType';
+import { ReferenceEventData } from '../domain/referenceEventData';
```
Comment (kspearrin) [inline src/models/request/registerRequest.ts:4]: can move this up with the other request model

Comment (cscharf) [reply src/models/request/registerRequest.ts]: Done

File: src/models/domain/referenceEventData.ts (hunk 1 above)
Comment (cscharf) [reply src/models/domain/referenceEventData.ts]: Done, hopefully 🤞

File: src/angular/components/register.component.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { PlatformUtilsService } from '../../abstractions/platformUtils.service';
 import { StateService } from '../../abstractions/state.service';
 
 import { KdfType } from '../../enums/kdfType';
+import { ReferenceEventRequest } from '../../models/request/referenceEventRequest';
```
Comment (kspearrin) [inline src/angular/components/register.component.ts:15]: you'll notice i try to keep all of these imports in a logical grouping. all request models are together, all services together, enums together, etc. move this up with the other request models.

Comment (cscharf) [reply src/angular/components/register.component.ts]: Done, sorry, I missed this one.

**Pull request merged**